src/charm.py

```diff
@@ -1300,8 +1335,45 @@ def _update_cert(self):
             # Repeat for the charm container.
             ca_cert_path.unlink(missing_ok=True)
 
-        container.exec(["update-ca-certificates", "--fresh"]).wait()
-        subprocess.run(["update-ca-certificates", "--fresh"])
+        self._update_system_certs()
+
+    def _on_trusted_certificate_available(self, event: CertificateAvailableEvent):
+        if not self.containers["workload"].can_connect():
+            logger.warning("Cannot connect to Pebble. Deferring event.")
+            event.defer()
+            return
+
+        self.restart_grafana()
+
+    def _update_trusted_ca_certs(self):
+        """This function receives the trusted certificates from the certificate_transfer integration."""
+        """Grafana needs to restart to use newly received certificates. Certificates attached to the
```

Comment:
Something went wrong here
```suggestion
        """This function receives the trusted certificates from the certificate_transfer integration.
        
        Grafana needs to restart to use newly received certificates. Certificates attached to the
```

Reply:
fixed